pkg/splunk/client/awss3client.go

```diff
@@ -64,20 +70,43 @@ func InitAWSClientWrapper(region, accessKeyID, secretAccessKey string) interface
 func InitAWSClientSession(region, accessKeyID, secretAccessKey string) SplunkAWSS3Client {
 	scopedLog := log.WithName("InitAWSClientSession")
 
+	// Enforcing minimum version TLS1.2
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			MinVersion: tls.VersionTLS12,
+		},
+	}
+	tr.ForceAttemptHTTP2 = true
+	httpClient := http.Client{Transport: tr}
+	disableSSL := false
+
 	sess, err := session.NewSession(&aws.Config{
 		Region: aws.String(region),
 		Credentials: credentials.NewStaticCredentials(
 			accessKeyID,     // id
 			secretAccessKey, // secret
 			""),
 		MaxRetries: aws.Int(3),
+		HTTPClient: &httpClient,
+		DisableSSL: &disableSSL,
 	})
 	if err != nil {
 		scopedLog.Error(err, "Failed to initialize an AWS S3 session.")
 		return nil
 	}
 
-	return s3.New(sess)
+	// Create the s3Client
+	s3Client := s3.New(sess)
+
+	// Validate transport
+	tlsVersion := "Unknown"
+	if tr, ok := s3Client.Config.HTTPClient.Transport.(*http.Transport); ok {
```

Comment:
what happens in case when ok=false? Are we ok with tlsVersion = Unknown?

Reply:
Since the tlsVersion is for informational purposes only, I think its ok that if this fails we get an "Unknown" TLS version.  I'll rerun the test with the default values and make sure.

Reply:
Also since we are explicitly setting the HTTPClient Transport above to enforce TLS1.2, this should always be ok.